aioesphomeapi/connection.py

```diff
@@ -246,24 +252,23 @@ def _cleanup(self) -> None:
             self._ping_timer = None
 
         if self.on_stop and self._connect_complete:
-
-            def _remove_on_stop_task(_fut: asyncio.Future[None]) -> None:
-                """Remove the stop task.
-
-                We need to do this because the asyncio does not hold
-                a strong reference to the task, so it can be garbage
-                collected unexpectedly.
-                """
-                self._on_stop_task = None
```

Comment:
I removed the closure here and bound it to the object since cython doesn't support cpdef's with closures.  We could take the cpdef out as well but it seemed better to remove the closure instead as its slower in python as well